### PR TITLE
Migrating test images to gcr.io/kubernetes-e2e-test-images

### LINKED ISSUE #62131: Migrate test images to gcr.io/kubernetes-e2e-test-images
/kind feature

**What happened**: Currently e2e test images are distributed between 2 different registry locations, k8s.gcr.io and gcr.io/kubernetes-e2e-test-images. https://github.com/kubernetes/kubernetes/blob/master/test/utils/image/manifest.go

**What you expected to happen**: We should ideally consolidate them under gcr.io/kubernetes-e2e-test-images where they rightly belong.

/assign

### PR DESCRIPTION
**Pull request by AishSundar**
**What this PR does / why we need it**:
Currently e2e test images are distributed between 2 different registry locations, k8s.gcr.io and gcr.io/kubernetes-e2e-test-images. This is part of a multi-step initiative to house all the images in gcr.io/kubernetes-e2e-test-images.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #62131

**Special notes for your reviewer**:
1. I am starting off with migrating images under k8s/test/images/volumes-tester. 
2. I did not move ceph and nfs images since they are marked for [deprecation and removal](https://github.com/kubernetes/kubernetes/tree/master/test/images/volumes-tester). Let me know if we want them moved as well.
3. I have made a copy of the images in gcr.io/kubernetes-e2e-test-images so the references are not broken post the PR merge. Will work on removing the images from k8s.gcr.io once this change sticks.

### CODE REVIEW

File: cluster/gce/manifests/e2e-image-puller.manifest [hunk 1]
```diff
@@ -70,10 +70,10 @@ spec:
       k8s.gcr.io/update-demo:kitten
       k8s.gcr.io/update-demo:nautilus
       k8s.gcr.io/volume-ceph:0.1
-      k8s.gcr.io/volume-gluster:0.2
-      k8s.gcr.io/volume-iscsi:0.1
+      gcr.io/kubernetes-e2e-test-images/volume-gluster:0.2
+      gcr.io/kubernetes-e2e-test-images/volume-iscsi:0.1
       k8s.gcr.io/volume-nfs:0.8
```
Comment (ixdy) [inline cluster/gce/manifests/e2e-image-puller.manifest]: maybe move this one, too?

**Comment by ixdy** [comment]
cc @mkumatag 
/ok-to-test

**Comment by AishSundar** [comment]
Looks like the 3 failing tests are flaky and failed in other presubmits as well
(i) https://k8s-testgrid.appspot.com/presubmits-kubernetes-nonblocking#pull-kubernetes-e2e-gke
(ii) https://k8s-testgrid.appspot.com/presubmits-kubernetes-scalability#pull-kubernetes-e2e-gce-100-performance
(iii) https://k8s-testgrid.appspot.com/presubmits-kubernetes-scalability#pull-kubernetes-kubemark-e2e-gce-big

Looking into the failures, they dont seem to be related to the change. @ixdy do I retry the tests and hope they pass for me to merge?

**Comment by AishSundar** [comment]
I reran the tests and all passed https://prow.k8s.io/pr?query=is%3Apr%20state%3Aopen%20author%3AAishSundar, except for pull-kubernetes-e2e-gce-100-performance which seems to be failing on a lot of PRs lately

https://k8s-testgrid.appspot.com/presubmits-kubernetes-scalability#pull-kubernetes-e2e-gce-100-performance

**Pull request merged**